Remove all references to @mdx-js/react

remcohaszing opened the pull request:
### What?

Remove all references to `@mdx-js/react`. One occurrence remains in the lockfile due to an indirect dependency.

### Why?

`@mdx-js/react` is actively harmful for Next.js users. `@mdx-js/react` uses React context. React context can’t be used in server components. As a result, users who have `@mdx-js/react` inside their `node_modules` get confusing errors.

Removing `@mdx-js/react` from all documentation, tests, and optional peer dependencies, has no drawbacks. Removing `@mdx-js/react` as a `next-mdx-import-source` alias, fixes the confusing errors.

Users who insist on using `@mdx-js/react`, can still do this by explicitly specifying the `providerImportSource` option.

### How?

Search for `@mdx-js/react`, and remove all occurences it.

ijjk commented:
Allow CI Workflow Run

- [ ] approve CI run for commit: f291d1f9973c4d173af7a7c9679413c0014a098e

Note: this should only be enabled once the PR is ready to go and can only be enabled by a maintainer

remcohaszing committed:
Remove all references to @mdx-js/react

`@mdx-js/react` is actively harmful for Next.js users. `@mdx-js/react`
uses React context. React context can’t be used in server components.
As a result, users who have `@mdx-js/react` inside their `node_modules`
get confusing errors.

Removing `@mdx-js/react` from all documentation, tests, and optional
peer dependencies, has no drawbacks. Removing `@mdx-js/react` as a
`next-mdx-import-source` alias, fixes the confusing errors.

Users who insist on using `@mdx-js/react`, can still do this by
explicitly specifying the `providerImportSource` option.

remcohaszing commented:
@icyJoseph Could you have a look?
